homeassistant/components/litterrobot/update.py

```diff
@@ -73,27 +59,20 @@ def in_progress(self) -> bool:
         """Update installation progress."""
         return self.robot.firmware_update_triggered
 
-    async def _async_update(self, _: HomeAssistant | datetime | None = None) -> None:
-        """Update the entity."""
-        self._poll_unsub = None
+    @property
+    def should_poll(self) -> bool:
+        """Set polling to True."""
+        return True
```

Comment:
@MartinHjelmare I had to define the `should_poll` property here because this inherits from a `CoordinatorEntity` which explicitly sets polling to false.

Reply:
Could you just set `_attr_should_poll` as `True`?

Reply:
No, because the CoordinatorEntity defines the property should_poll and returns False in it. So it never looks at the attribute.

See https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/update_coordinator.py#L360

Reply:
Unless you want the update entity to also update its state when the coordinator fetches new data, I suggest not inheriting the coordinator entity here.

Reply:
In this scenario, both are desired. The coordinator entity is handling the main refresh/update of the robot, which includes the firmware version. The polling is hitting a different endpoint to check if there is new firmware, which is only needed for this entity.

Reply:
This entity doesn't seem to use any data from the coordinator. Did I miss it?

Reply:
Is it the `self.robot.firmware`?

Reply:
> Is it the `self.robot.firmware`?

Yes, it is

Reply:
Ok. :+1: